grouper/models.py

```diff
@@ -1680,6 +1627,48 @@ def my_log_entries(self):
         return AuditLog.get_entries(self.session, on_permission_id=self.id, limit=20)
 
 
+class PermissionRequest(Model):
+    """Represent request for a permission/argument to be granted to a particular group."""
+    __tablename__ = "permission_requests"
+
+    id = Column(Integer, primary_key=True)
+
+    # The User that made the request.
+    requester_id = Column(Integer, ForeignKey("users.id"), nullable=False)
+    requester = relationship(
+        User, backref="permission_requests", foreign_keys=[requester_id]
+    )
+
+    permission_id = Column(Integer, ForeignKey("permissions.id"), nullable=False)
+    permission = relationship(Permission, foreign_keys=[permission_id])
+    argument = Column(String(length=64), nullable=True)
+
+    group_id = Column(Integer, ForeignKey("groups.id"), nullable=False)
+    group = relationship(Group, foreign_keys=[group_id])
+
+    requested_at = Column(DateTime, default=datetime.utcnow, nullable=False)
+
+    status = Column(Enum(*REQUEST_STATUS_CHOICES), default="pending", nullable=False)
+
+
+class PermissionRequestStatusChange(Model):
+    """Tracks changes to each permission grant request."""
+    __tablename__ = "permission_request_status_changes"
+
+    id = Column(Integer, primary_key=True)
+
+    request_id = Column(Integer, ForeignKey("permission_requests.id"), nullable=False)
+    request = relationship(PermissionRequest)
```

Comment:
¯\_(ツ)_/¯

the membership request schema doesn't include it either. but mostly because we haven't needed to build the functionality of getting back from a random comment to a request. i suppose if we ever build a comment viewer it makes sense but i'm inclined to leave it off for now.